src/main/java/com/google/devtools/build/lib/vfs/Path.java

```diff
@@ -236,6 +237,21 @@ public int compareTo(Path o) {
     return pathFragment.compareTo(o.pathFragment);
   }
 
+  /**
+   * Returns the same path on the underlying file system if the current file system is a partial
+   * overlay of another file system. Otherwise, returns the current path unchanged.
+   *
+   * <p>Note that the returned path may still reference an in-memory file system (in tests, for
+   * example), but should be treated as being on the "native" file system for the host machine.
+   */
+  public Path onUnderlyingFileSystem() {
```

Comment:
Nit: let's call it `for` instead of `on`? The former is more common for factory methods, and the latter sounds like an event (to me at least).